### PR TITLE
#167142660 create icon atom

### PR DESCRIPTION
**Pull request by Sojisoyoye**
#### What does this PR do?

This PR creates icon field atom

#### Description of task to be completed?

- write test for the feature
- create style component for icon field
- create icon field component

#### How should this be manually tested?

- run npm install
- run npm run styleguide on your console

#### Any background context you want to provide?

-  install node

#### What are the relevant pivotal tracker stories?

[167142660](https://www.pivotaltracker.com/n/projects/2351184/stories/167142660)

#### Screenshots (if appropriate)

image

### CODE REVIEW

**Review by akinyeleolat — APPROVED** [review]

**Review by mosinmiloluwa01 — APPROVED** [review]

File: src/components/UI/atoms/Icon/Icon.jsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+const Icon = ({
+  id,
+  color,
+  height,
+  width,
+  display,
+  margin,
+  icon,
+}) => {
+  const IconType = icomoon[icons[icon]];
```
Comment (benfluleck) [inline src/components/UI/atoms/Icon/Icon.jsx]: what is `icomoon`?

File: src/components/UI/atoms/Icon/Icon.jsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+ * @return {component} Icon
+ */
+const Icon = ({
+  id,
+  color,
+  height,
+  width,
+  display,
+  margin,
+  icon,
```
Comment (benfluleck) [inline src/components/UI/atoms/Icon/Icon.jsx]: Please change to iconName I think that is better

File: src/components/UI/atoms/Icon/Icon.jsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  color: PropTypes.oneOf(Object.keys(iconColors)),
+  icon: PropTypes.oneOf(Object.keys(icons)).isRequired,
+  height: PropTypes.oneOf(Object.keys(iconSizes)),
+  width: PropTypes.oneOf(Object.keys(iconSizes)),
+  display: PropTypes.oneOf(['block', 'inline', 'inline-block', 'none']),
+  margin: PropTypes.oneOf(Object.keys(spacing)),
+};
+
+Icon.defaultProps = {
+  id: 'mockIcon',
```
Comment (benfluleck) [inline src/components/UI/atoms/Icon/Icon.jsx]: why would you use this as a defaultProp name ?

File: src/components/UI/atoms/Icon/Icon.jsx [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  icon: 'search',
```
Comment (benfluleck) [inline src/components/UI/atoms/Icon/Icon.jsx]: Why would you want your default iconName be search

File: src/components/UI/atoms/Icon/Icon.spec.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  it('should render the Icon component with default prop', () => {
+    const { container } = setup();
+
+    expect(container).toBeTruthy();
+  });
+
+  it('should render the Icon with height', () => {
+    const { getByTestId } = setup({ height: 'medium' });
+
+    const icon = getByTestId('mockIcon');
```
Comment (benfluleck) [inline src/components/UI/atoms/Icon/Icon.spec.js]: Use queryByTestId https://testing-library.com/docs/dom-testing-library/api-queries

**Review by benfluleck — CHANGES_REQUESTED** [review]

File: src/components/UI/atoms/Icon/Icon.jsx (hunk 1 above)
Comment (Sojisoyoye) [reply src/components/UI/atoms/Icon/Icon.jsx]: icomoon is the name of the icon pack from styled-icon component library.

Comment (benfluleck) [reply src/components/UI/atoms/Icon/Icon.jsx]: please rename since it is a default export for readability

Comment (Sojisoyoye) [reply src/components/UI/atoms/Icon/Icon.jsx]: ok

File: src/components/UI/atoms/Icon/Icon.jsx (hunk 4 above)
Comment (Sojisoyoye) [reply src/components/UI/atoms/Icon/Icon.jsx]: The component requires that an iconName is supplied for it to render, as it is a required prop-type and search is one of the icons we need on the homepage so I decided to use it as default.

**Pull request merged**